tools/validate_spec.py

```diff
@@ -0,0 +1,314 @@
+#!/usr/bin/env python3
```

Comment:
IMHO adding a small set of integration tests would make sense to test the correct output and exit code in different scenarios.

Reply:
Added in 801738c, 44e512e